tket/src/Circuit/macro_manipulation.cpp

```diff
@@ -248,16 +249,30 @@ void Circuit::substitute(
                  // trivial, essentially rewiring on a cut
   std::map<Edge, Vertex> c_out_map;
 
+  std::set<Qubit> reset_qbs;
+  for (auto qb : to_insert.all_qubits()) {
```

Comment:
`const Qubit& qb` - or is there a reason to not pass by constant reference/reference here?

Reply:
Done.